src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java

```diff
@@ -1,62 +1,115 @@
 package uk.gov.hmcts.reform.cwrdapi.servicebus;
 
+import com.azure.messaging.servicebus.ServiceBusMessage;
+import com.azure.messaging.servicebus.ServiceBusMessageBatch;
+import com.azure.messaging.servicebus.ServiceBusSenderClient;
+import com.azure.messaging.servicebus.ServiceBusTransactionContext;
+import com.google.gson.Gson;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.ListUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.jms.core.JmsTemplate;
-import org.springframework.retry.annotation.Backoff;
-import org.springframework.retry.annotation.Recover;
-import org.springframework.retry.annotation.Retryable;
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.cwrdapi.client.domain.PublishCaseWorkerData;
+import uk.gov.hmcts.reform.cwrdapi.config.MessagingConfig;
 import uk.gov.hmcts.reform.cwrdapi.controllers.advice.CaseworkerMessageFailedException;
 import uk.gov.hmcts.reform.cwrdapi.service.IValidationService;
 import uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 import javax.validation.constraints.NotNull;
 
 @Slf4j
 @Service
 public class TopicPublisher {
 
-    private final JmsTemplate jmsTemplate;
-    private final String destination;
     private final IValidationService validationService;
+    private final MessagingConfig messagingConfig;
 
     @Value("${loggingComponentName}")
     private String loggingComponentName;
 
+    @Value("${crd.publisher.caseWorkerDataPerMessage}")
+    private int caseWorkerDataPerMessage;
+
+    @Value("${crd.publisher.azure.service.bus.topic}")
+    private String topic;
+
     @Autowired
-    public TopicPublisher(JmsTemplate jmsTemplate,
-                          @Value("${crd.publisher.azure.service.bus.topic}") final String destination,
-                          IValidationService validationService) {
-        this.jmsTemplate = jmsTemplate;
-        this.destination = destination;
+    public TopicPublisher(IValidationService validationService, MessagingConfig messagingConfig) {
         this.validationService = validationService;
+        this.messagingConfig = messagingConfig;
     }
 
-    @Retryable(maxAttempts = 3, backoff = @Backoff(delay = 2000, multiplier = 3))
-    public void sendMessage(@NotNull Object message) {
-        log.info("{}:: Publishing message to service bus topic:: Job Id is: {}", loggingComponentName,
-                validationService.getAuditJobId());
-        if (message instanceof PublishCaseWorkerData) {
+    //The Azure library has retry mechanism inbuilt, which will try to send message until timeout.
+    // There is no need for explicit retry mechanism.
+    public void sendMessage(@NotNull PublishCaseWorkerData caseWorkerIds) {
+        ServiceBusSenderClient serviceBusSenderClient = null;
```

Comment:
you can Autowire the ServiceBusSenderClient object (once you create the  bean ServiceBusSenderClient)and use it like
`@Autowired
    ServiceBusSenderClient serviceBusSenderClient;`

Reply:
Hi @SujeethKumarAri @abhijitdiwan I have created the bean for service bus client. Can you please have a look.
Thanks